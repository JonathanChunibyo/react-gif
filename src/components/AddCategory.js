import React, { useState } from 'react'
import PropTypes from "prop-types";


export const AddCategory = ({ setcategory }) => {
    const vacio = () => {
        setinputValue('');
    }
    const [inputValue, setinputValue] = useState('');
    const handleInputvalue = (e) => {
        setinputValue(e.target.value);
    }
    const handleSubmit = (e) => {
        e.preventDefault();
        if (inputValue.trim().length > 2) {
            setcategory(cats => [inputValue, ...cats]);
            setinputValue('');
        }
    }
    return (

        <form onSubmit={handleSubmit}>

            <h1> {inputValue}</h1>
            <input
                type="text"
                value={inputValue}
                onChange={handleInputvalue}
                onClick={vacio}
            />
        </form>
    )
}

AddCategory.propTypes = {
    setcategory: PropTypes.func.isRequired ,
}

