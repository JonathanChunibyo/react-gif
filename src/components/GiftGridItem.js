import React from 'react';
import PropTypes from 'prop-types';

export const GiftGridItem = ({title, url}) => {
    return (
        <div className='card animate__animated animate__fadeInLeft'>
            <p>{title}</p>
            <img src= {url} alt={title}/>
        </div>
    )
}
GiftGridItem.propTypes = {
    title: PropTypes.string.isRequired ,
    url: PropTypes.string.isRequired,
}
