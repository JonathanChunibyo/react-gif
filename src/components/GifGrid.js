import React /* { useEffect, useState } */ from 'react'
import { useFetchGif } from '../hooks/useFetchGif'
//import { getGif } from '../helpers/getGifs';
import { GiftGridItem } from './GiftGridItem';
import PropTypes from "prop-types";


export const GifGrid = ({ category }) => {


    const {data: images, loading} = useFetchGif(category);


    
    
    /* const [images, setimages] = useState([]);
    useEffect(() => {
        getGif(category).then( setimages );
    }, [category]);
 */

    return (
        <>

            <h3 className='animate__animated animate__zoomInDown'> {category} </h3>
            {loading && <p className='animate__animated animate__shakeX'>'Cargando...'</p>}
            
            <div className='cardgrid'>
                {
                     //images.map(img => (
                       // <li key={img.id}>{img.title}</li>
                    //)) 
                    images.map((img) =>
                        <GiftGridItem
                            key={img.id}
                            {...img}
                        />)
                }
            </div>
        </>
    )
}
GifGrid.propTypes = {
    category: PropTypes.string.isRequired,
}