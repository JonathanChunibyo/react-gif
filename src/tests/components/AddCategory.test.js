import { shallow } from "enzyme"
import React from 'react';
import { AddCategory } from "../../components/AddCategory"


describe('pruebas <AppCategory />', () => {

    const setCategory = jest.fn();
    let wraper = shallow(<AddCategory setcategory={setCategory} />);
    const value = 'mundo';

    beforeEach(() => {
        jest.clearAllMocks();
        wraper = shallow(<AddCategory setcategory={setCategory} />);
    })

    test('debe mostrarse correctamente ', () => {

        expect(wraper).toMatchSnapshot();
    })
    test('debe cambiar la caja de texto', () => {

        const input = wraper.find('input');
        input.simulate('change', { target: { value } });
    })
    test('No debe almacenar info con submit', () => {

        wraper.find('form').simulate('submit', { preventDefault() { } });
        expect(setCategory).not.toHaveBeenCalled();

    })
    test('vaciar caja de texto y llamar el setCategory', () => {
        const input = wraper.find('input');
    
        //simular input escritura
        input.simulate('change', { target: { value } });
        //simular enviar formulario
        wraper.find('form').simulate('submit', { preventDefault() { } });

        // saber si el setCategory fue llamado
        expect(setCategory).toHaveBeenCalled();
        /* 
        llamar como si fuera una funcion
        expect(setCategory).toHaveBeenCalledWith( expect.any(Function) );
 */
        //el valor del input debe estar en ''
 
        expect( wraper.find('input').prop('value')).toBe('');
        //nose lo hice yo ;v
        expect(input.text().trim()).toBe('');
    })
    



})
