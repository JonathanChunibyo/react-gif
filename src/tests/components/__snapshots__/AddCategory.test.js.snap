// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas <AppCategory /> debe mostrarse correctamente  1`] = `
<form
  onSubmit={[Function]}
>
  <h1>
     
  </h1>
  <input
    onChange={[Function]}
    onClick={[Function]}
    type="text"
    value=""
  />
</form>
`;
